common/ledger/blkstorage/blockstore_provider.go

```diff
@@ -125,6 +125,29 @@ func (p *BlockStoreProvider) Exists(ledgerid string) (bool, error) {
 	return exists, err
 }
 
+// Remove block index and blocks for the given ledgerid (channelID). It is not an error if the channel does not exist.
+// This function is not error safe. If this function returns an error or a crash takes place, it is highly likely
+// that the data for this ledger is left in an inconsistent state. Opening the ledger again or reusing the previously
+// opened ledger can show unknown behavior.
+func (p *BlockStoreProvider) Remove(ledgerid string) error {
+	exists, err := p.Exists(ledgerid)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return nil
+	}
+	dbHandle := p.leveldbProvider.GetDBHandle(ledgerid)
+	if err := dbHandle.DeleteAll(); err != nil {
+		return err
+	}
+	dbHandle.Close()
```

Comment:
Nit. Can be deferred after getting the handle.

Reply:
`dbHandle.Close()` removes dbHandle from provider's dbHandles map, so it is only called after `DeleteAll` is successful.

Reply:
I meant `defer dbHandle.Close()` at line 140.